spec/grover/processor_spec.rb

```diff
@@ -579,6 +603,46 @@
         it { expect(pdf_text_content).to eq "#{date} http://www.example.net/foo/bar 1/1" }
       end
 
+      context 'when wait for function option is specified with options' do
+        let(:url_or_html) do
+          <<-HTML
+            <html>
+              <body></body>
+
+              <script>
+                setTimeout(function() {
+                  document.body.innerHTML = '<h1 id="test">Hey there</h1>';
+                }, 500);
+              </script>
+            </html>
+          HTML
+        end
+        let(:options) do
+          basic_header_footer_options.merge(
+            'waitForFunction' => 'document.getElementById("test") !== null',
+            'waitForFunctionOptions' => { "polling": 50, "timeout": 750 }
+          )
+        end
+        let(:date) { Date.today.strftime '%-m/%-d/%Y' }
+
+        it { expect(pdf_text_content).to eq "#{date} Hey there http://www.example.net/foo/bar 1/1" }
+
+        context 'when waiting for function takes too long' do
+          let(:options) do
+            basic_header_footer_options.merge(
+              'waitForFunction' => 'document.getElementById("test") !== null',
+              'waitForFunctionOptions' => { "polling": 50, "timeout": 250 }
+            )
+          end
+
+          it 'raises a JavaScript error if waitForFunction fails' do
+            expect do
+              pdf_text_content
+            end.to raise_error Grover::JavaScript::Error
```

Comment:
suggest also adding either a string match for the message, or if that gets complicated, a regex match. ie: 
```
end.to raise_error Grover::JavaScript::Error, 'The full error message'
```
or

```
end.to raise_error Grover::JavaScript::Error, /partial match/
```
I would expect the match to be something about it being a timeout. Just makes following the intent of the test a little clearer

Reply:
Updated this to be `Grover::JavaScript::TimeoutError, /waiting for function failed/`